model/articlesModel.js

```diff
@@ -10,15 +10,21 @@ exports.selectArticleById = (article_id) => {
     });
 };
 
-exports.selectArticles = (sort_by = 'created_at', order = 'DESC') => {
+exports.selectArticles = (sort_by = 'created_at', order = 'DESC', topic) => {
   let queryString = `SELECT articles.*,
     COUNT(comments.comment_id) AS comment_count
     FROM articles
-    LEFT JOIN comments ON articles.article_id = comments.article_id
-    GROUP BY
-    articles.article_id `;
+    LEFT JOIN comments ON articles.article_id = comments.article_id `;
   const validSortBy = ['title', 'author', 'topic', 'votes', 'created_at'];
   const validOrder = ['ASC', 'DESC'];
+  const validTopics = ['cats', 'mitch', 'paper'];
+
+  if (topic && validTopics.includes(topic)) {
```

Comment:
Better to make the check for valid topics dynamic by checking in the database as this could break with the dev data currently. Really nice logic building up your query string though!